### ISSUE TITLE
Host Deletion function performs no check for the type of Host

### ISSUE THREAD

**Issue by Kajol-Kumari** [Cloud-CV/EvalAI#2664]
#Bug

Expected behaviour:
Admin should not be allowed to delete himself.

Observed behaviour:
When Admin tries to remove himself, the whole team along with his members get removed.
 
Steps to reproduce the problem:
After writing controller function for deleting host.
Go to Host Team -> Create a new host team -> Add members to it -> Try deleting Admin 

Operating System:
Ubuntu 18.04

Browser:
Google Chrome

![ezgif com-video-to-gif (6)](https://user-images.githubusercontent.com/44888949/74369667-c5453a80-4dfb-11ea-8b68-ac1c80c59be6.gif)

Reference: https://github.com/Cloud-CV/EvalAI/blob/master/apps/hosts/views.py#L204-L206

**Comment by Suryansh5545** [comment]
@Kajol-Kumari As you have added this new feature and you know what possibly went wrong would to like to fix it?

**Comment by Suryansh5545** [comment]
@Kajol-Kumari The feature isn't merged yet so you can fix this bug in that and close this issue.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari The feature isn't merged yet so you can fix this bug in that and close this issue.

@Suryansh5545  The part of code i have given in reference is already merge. While working on deletion feature of host, i found that the codes for deletion is already written but it is not performing any check on the type of  user trying to delete a particular host.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari As you have added this new feature and you know what possibly went wrong would to like to fix it?

I am working on it @Suryansh5545

**Comment by deshraj** [comment]
@Kajol-Kumari we have an updated plan for this.

- Only person who created the team should be able to delete members from the team and invited members should not have the rights to delete other users in the team. This applies to both participants and hosts team.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari we have an updated plan for this.
> 
> * Only person who created the team should be able to delete members from the team and invited members should not have the rights to delete other users in the team. This applies to both participants and hosts team.

Yes @deshraj . This is what i have implemented [here](https://github.com/Cloud-CV/EvalAI/pull/2781/files#diff-936afd091c6c99c1a02293abc3be8653R201). Invited members won't have the privilege to delete any of the members and also the `delete` icon would not be visible to him/her. Please see the 3rd GIF [here](https://github.com/Cloud-CV/EvalAI/pull/2781)
